### PR TITLE
Get statistics on tools usage per language / compiler

### PR DESCRIPTION
**Pull request by frederic-tingaud-sonarsource**
As discussed with Matt, adding statistics at the tool level could give precious information to tool developers.

### CODE REVIEW

**Comment by frederic-tingaud-sonarsource** [comment]
I made sure the code and test still run well after that, but I didn't actually test that statistics are well gathered because I'm not sure how to access them.

**Comment by partouf** [comment]
If you pass ‘—metricsPort 10241’ to node, you’ll be able to see them in your browser when you go to 127.0.0.1:10241

File: lib/tooling/base-tool.js [hunk 1]
```diff
@@ -24,19 +24,31 @@
...
     constructor(toolInfo, env) {
         this.tool = toolInfo;
         this.env = env;
         this.tool.exclude = this.tool.exclude ? this.tool.exclude.split(':') : [];
         this.addOptionsToToolArgs = true;
         this.parseOutput = utils.parseOutput;
+
+        const toolName = this.tool.name;
+        if (toolName && !(toolName in countersPerTool)) {
+            countersPerTool[toolName] = new PromClient.Counter({
```
Comment (mattgodbolt) [inline lib/tooling/base-tool.js:46]: Pinging @apmorton on this - do you think we should use a single counter with different labels? Else we have no way to do a dashboard with "all tool invocations" without listing all the possible names of tools (IIUC).

File: lib/tooling/base-tool.js [hunk 2]
```diff
@@ -24,19 +24,31 @@
...
+                name: `${toolName.replaceAll(/\W/g, '_').toLowerCase()}_compilation_total`,
+                help: `Number of compilations with ${toolName} active`,
+                labelNames: ['language', 'compiler'],
```
Comment (mattgodbolt) [inline lib/tooling/base-tool.js:49]: I'm not sure the `compiler` is useful here? Tools run independently of the compiler (I think?)

File: lib/tooling/base-tool.js (hunk 1 above)
Comment (mattgodbolt) [reply lib/tooling/base-tool.js:46]: No worries, it was easy enoug to fix and I already updated them! :) Sorry I should have posted here! #3883

**Pull request merged**